service/matching/handler/engine.go

```diff
@@ -149,15 +153,18 @@ func NewEngine(taskManager persistence.TaskManager,
 		partitioner:          partitioner,
 		timeSource:           timeSource,
 	}
+
+	go e.subscribeToMembershipChanges()
+
 	e.waitForQueryResultFn = e.waitForQueryResult
 	return e
 }
 
 func (e *matchingEngineImpl) Start() {
-	// As task lists are initialized lazily nothing is done on startup at this point.
 }
 
 func (e *matchingEngineImpl) Stop() {
+	close(e.shutdown)
```

Comment:
we're not waiting for `subscribeToMembershipChanges` to complete, possibly goroutine leak?

Reply:
This is true: I was too lazy and didn't think to add a full WG setup to engine. What do you think? It does mean that I didn't toggle it on for the leak-detector

Reply:
+1 for not leaving goroutines behind. let's wait here until `subscribeToMembershipChanges` returns (via waitgroup)

Reply:
Ok, I don't love adding a bunch of complexity *back* to the shutdown process, but fair. Added a waitgroup.